sabre1041/managing-ocp-install-beyond
# Incorporation of APB framework

Several changes are part of this PR:

* Addition of an Ansible Playbook Bundle (APB) to deploy the alertmanager-tower-bridge
* Support for workflow jobs in bridge application
* Minor Prometheus corrections

To Test:

1. Pull down changes included in this branch and deploy a brand new lab environment
2. Open Ansible Tower UI in web browser
    1. Go to _Projects_ tab and edit _Managing OCP from Install and Beyond_ project
    2. Change scm/branch to `ablock-apb-framework`
    3. Make sure `Clean` and `Delete on Update` are checked
    4. Click **Save**
    5. Trigger update of  _Managing OCP from Install and Beyond_ project
3. Navigate to **Templates** tab and launch **2-Scaleup_OpenShift_on_AWS** job template
    1. Wait for workflow job to complete
4. Deploy the ABP
    1. Login to the Master
    2. Sudo up: `sudo -i`
    3. Install apb `yum install -y apb`
    4. Give rights to the user: `oc adm policy add-cluster-role-to-user cluster-admin <student-name>`
    5.  Login as user: `oc login -u <student-name>`
    6. Clone the branch on the master host
    7.  Navigate the `alertmanager-tower-bridge-apb` directory
    8. Build the apb: `apb build`
    9. Push the apb to the service catalog: `apb push`
5. Provision the Tower Bridge Application
    1. Launch the OpenShift web UI and log into as <student-name>
    2. Change into the _openshift-metrics_ project
    3. Click **Add to Project** and then **Browse Catalog**
    4. Select _AlertManager Ansible Tower Bridge_ from catalog
        1. Tower Host: Enter the hostname for Ansible tower (ie: tower-ablock-1.labs.sysdeseng.com)
        2. Enter and confirm the password for Tower
        3. Update the name of the Git Branch to `ablock-apb-framework`
        4. Click **Create**
6. Confirm the buildconfig, service, secret, deploymentconfig have been created, a build has completed and the bridge application deployed
7. To verify deprovisioning, click on the kabbob next to the provisioned service on the overview page, and click **Delete**. The resources will be removed
